photosynthesis_metrics/utils.py

```diff
@@ -0,0 +1,35 @@
+import torch
+
+from typing import Optional, Union, Tuple, List
+def _adjust_dimensions(x: torch.Tensor, y: torch.Tensor):
+    r"""Expands input tensors dimensions to 4D
+    """
+    # TODO: try to move this block in __compute_ssim since it is very general.
+    # TODO: add support of 5D tensors here or in the __compute_ssim function.
+    num_dimentions = x.dim()
+    if num_dimentions == 2:
+        x = x.expand(1, 1, *x.shape)
+        y = y.expand(1, 1, *y.shape)
+    elif num_dimentions == 3:
+        x = x.expand(1, *x.shape)
+        y = y.expand(1, *y.shape)
+    elif num_dimentions != 4:
+        raise ValueError('Expected 2, 3, or 4 dimensions (got {})'.format(num_dimentions))
+
+    return x, y
+
+
+def _validate_input(x: torch.Tensor, y: torch.Tensor, kernel_size: Optional[int] = None,
```

Comment:
Please consider making `y: Optional[torch.Tensor] = None`. It will remove the need for you to do things like `y=x` in your total variance interface.